MinecraftClient/Protocol/Handlers/Protocol18.cs

```diff
@@ -1367,7 +1367,39 @@ private static ushort[] readNextUShortsLittleEndian(int amount, List<byte> cache
         /// <returns>The uuid</returns>
         private static Guid readNextUUID(List<byte> cache)
         {
-            return new Guid(readData(16, cache));
+            byte[] javaUUID = readData(16, cache);
+            Guid guid;
+            if (BitConverter.IsLittleEndian)
+            {
+                guid = ToLittleEndian(javaUUID);
+            }
+            else
+            {
+                guid = new Guid(javaUUID);
+            }
+            return guid;
+        }
+
+        /// <summary>
+        /// convert a Java big-endian Guid to a .NET little-endian Guid.
+        /// </summary>
+        /// <returns>GUID in little endian.</returns>
+        private static Guid ToLittleEndian(byte[] java)
```

Comment:
I'm confident there are better and smarter versions on how do to that conversion in C#. In all honesty I just took the first [example](https://stackoverflow.com/a/10149822/7890667) that worked :P